internal/cmd/controller/target/status_test.go

```diff
@@ -0,0 +1,504 @@
+// unavailable due to a mismatched ID.
+func unavailableTargetMismatchedID() *Target {
+	return &Target{
+		Deployment: &fleet.BundleDeployment{
+			Spec: fleet.BundleDeploymentSpec{
+				DeploymentID:       "id",
+				StagedDeploymentID: "id",
+			},
+			Status: fleet.BundleDeploymentStatus{
+				AppliedDeploymentID: "off-id",
+				Ready:               true,
+			},
+		},
+		DeploymentID: "id",
+	}
+}
+
+// unavailableTargetNonReady returns a Target that is considered
+// unavailable due to not being ready.
+func unavailableTargetNonReady() *Target {
+	return &Target{
+		Deployment: &fleet.BundleDeployment{
+			Spec: fleet.BundleDeploymentSpec{
+				DeploymentID:       "id",
+				StagedDeploymentID: "id",
+			},
+			Status: fleet.BundleDeploymentStatus{
+				AppliedDeploymentID: "id",
+				Ready:               false,
+			},
+		},
+		DeploymentID: "id",
+	}
+}
+
+func targetWithRolloutStrategy(target *Target, rolloutStrategy fleet.RolloutStrategy) *Target {
+	if target.Bundle == nil {
+		target.Bundle = &fleet.Bundle{}
+	}
+	target.Bundle.Spec.RolloutStrategy = &rolloutStrategy
+	return target
+}
+
+func Test_limit(t *testing.T) {
+	tests := []struct {
+		name    string
+		count   int
+		val     []*intstr.IntOrString
+		want    int
+		wantErr bool
+	}{
+		{
+			name:  "fixed value below count",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{IntVal: 5},
+			},
+			want: 5,
+		},
+		{
+			name:  "fixed value above count",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{IntVal: 15},
+			},
+			want: 15,
+		},
+		{
+			name:  "with value with zero count",
+			count: 0,
+			val: []*intstr.IntOrString{
+				{IntVal: 15},
+			},
+			want: 1,
+		},
+		{
+			name:  "fixed value with negative count",
+			count: -15,
+			val: []*intstr.IntOrString{
+				{IntVal: 15},
+			},
+			want: 1,
+		},
+		{
+			name:  "two fixed values should take the first one",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{IntVal: 5},
+				{IntVal: 15},
+			},
+			want: 5,
+		},
+		{
+			name:  "two fixed values should ignore nil",
+			count: 10,
+			val: []*intstr.IntOrString{
+				nil,
+				{IntVal: 15},
+			},
+			want: 15,
+		},
+		{
+			name:  "percent value 50",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{Type: intstr.String, StrVal: "50%"},
+			},
+			want: 5,
+		},
+		{
+			name:  "percent value 10",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{Type: intstr.String, StrVal: "10%"},
+			},
+			want: 1,
+		},
+		{
+			name:  "negative percent value",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{Type: intstr.String, StrVal: "-10%"},
+			},
+			want: 1,
+		},
+		{
+			name:  "percent value 10 with count 5",
+			count: 5,
+			val: []*intstr.IntOrString{
+				{Type: intstr.String, StrVal: "10%"},
+			},
+			want: 1,
+		},
+		{
+			name:  "no value should match count",
+			count: 50,
+			val:   []*intstr.IntOrString{},
+			want:  50,
+		},
+		{
+			name:  "percentage below 1 should return 1",
```

Comment:
Isn't this the same test case as `percent value 10 with count 5`?

Reply:
Yes! Removed it.